File: integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/GenericEntity.java
package io.quarkus.it.panache.reactive;

import javax.persistence.MappedSuperclass;

import io.quarkus.hibernate.reactive.panache.PanacheEntity;

@MappedSuperclass
public abstract class GenericEntity<T> extends PanacheEntity {
    public T t;
    public T t2;

    public T getT2() {
        return t2;
    }

    public void setT2(T t2) {
        this.t2 = t2;
    }
}

File: integration-tests/hibernate-reactive-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/reactive/kotlin/GenericEntity.kt
package io.quarkus.it.panache.reactive.kotlin

import io.quarkus.hibernate.reactive.panache.kotlin.PanacheEntity
import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class GenericEntity<T> : PanacheEntity() {
    var t: T? = null
    var t2: T? = null
}